geopandas/tests/test_plotting.py

```diff
@@ -1731,6 +1731,29 @@ def compare_figures(self, kind, fig_test, fig_ref, kwargs):
     if MPL_DECORATORS:
 
         @pytest.mark.parametrize("kind", _pandas_kinds)
+        @pytest.mark.filterwarnings(
+            ""
+            if compat.PANDAS_GE_15
+            else (
+                "ignore:The get_cmap function will be deprecated in a future version:"
+                "PendingDeprecationWarning:pandas.*"
+            )
+        )
+        @pytest.mark.filterwarnings(
+            ""
+            if compat.PANDAS_GE_15
+            else "ignore:No data for colormapping provided:UserWarning:pandas.*"
+        )
+        @pytest.mark.filterwarnings(
+            ""
+            if compat.PANDAS_GE_12
+            else "ignore::DeprecationWarning:matplotlib.*|fontTools.*"
+        )
+        @pytest.mark.filterwarnings(
+            ""
+            if compat.PANDAS_GE_12
+            else "ignore::Warning:pandas.plotting._matplotlib.*:331"
+        )
```

Comment:
BTW, I wouldn't care too much about this kind of warnings that only happen on builds with old versions (and in this case also only because we should also pick an older version of matplotlib that matches the lifetime of the pandas version)

And to be clear, it's super valuable to get the warnings under control so we can more easily the warnings we actually should address (I did similar PRs in the past as well, your work is very much appreciated!). But I would focus on the warning that appear in the "latest" + "dev" builds.

Reply:
there are 2 other PRs #2645 and #2642 where I'm deliberately reducing warnings from **pandas** to zero.  Easier to work with none rather than a small number that can be ignored ....  my next step is another branch / PR focuses on warnings coming from **array.py**.   locally I'm merging these branches to see overall impact, from a push perspective looking to keep them bite sized for simpler merge into main

Reply:
another thing I've thought about but haven't really progressed is a better way for structuring env YaML files so that versions are more compatible for backwards compatibility testing.   for that I'm trying to get https://github.com/nektos/act working but it's horribly slow!  may fork it and try to get it to run on a configured vagrant vm instead of docker